src/Catalog/Helpers/Db2CatalogProjection.cs

```diff
@@ -19,29 +22,58 @@ public Db2CatalogProjection(PackageContentUriBuilder packageContentUriBuilder)
             {
-                throw new ArgumentNullException(nameof(dataRecord));
+                throw new ArgumentNullException(nameof(dataReader));
             }
 
-            var packageId = dataRecord[Db2CatalogProjectionColumnNames.PackageId].ToString();
-            var normalizedPackageVersion = dataRecord[Db2CatalogProjectionColumnNames.NormalizedVersion].ToString();
-            var listed = dataRecord.GetBoolean(dataRecord.GetOrdinal(Db2CatalogProjectionColumnNames.Listed));
-            var hideLicenseReport = dataRecord.GetBoolean(dataRecord.GetOrdinal(Db2CatalogProjectionColumnNames.HideLicenseReport));
+            var packageId = dataReader[Db2CatalogProjectionColumnNames.PackageId].ToString();
+            var normalizedPackageVersion = dataReader[Db2CatalogProjectionColumnNames.NormalizedVersion].ToString();
+            var listed = dataReader.GetBoolean(dataReader.GetOrdinal(Db2CatalogProjectionColumnNames.Listed));
+            var hideLicenseReport = dataReader.GetBoolean(dataReader.GetOrdinal(Db2CatalogProjectionColumnNames.HideLicenseReport));
 
             var packageContentUri = _packageContentUriBuilder.Build(packageId, normalizedPackageVersion);
+            var deprecationInfo = ReadDeprecationInfoFromDataRecord(dataReader);
 
             return new FeedPackageDetails(
                 packageContentUri,
-                dataRecord.ReadDateTime(Db2CatalogProjectionColumnNames.Created).ForceUtc(),
-                dataRecord.ReadNullableUtcDateTime(Db2CatalogProjectionColumnNames.LastEdited),
-                listed ? dataRecord.ReadDateTime(Db2CatalogProjectionColumnNames.Published).ForceUtc() : Constants.UnpublishedDate,
+                dataReader.ReadDateTime(Db2CatalogProjectionColumnNames.Created).ForceUtc(),
+                dataReader.ReadNullableUtcDateTime(Db2CatalogProjectionColumnNames.LastEdited),
+                listed ? dataReader.ReadDateTime(Db2CatalogProjectionColumnNames.Published).ForceUtc() : Constants.UnpublishedDate,
                 packageId,
                 normalizedPackageVersion,
-                hideLicenseReport ? null : dataRecord[Db2CatalogProjectionColumnNames.LicenseNames]?.ToString(),
-                hideLicenseReport ? null : dataRecord[Db2CatalogProjectionColumnNames.LicenseReportUrl]?.ToString());
+                hideLicenseReport ? null : dataReader[Db2CatalogProjectionColumnNames.LicenseNames]?.ToString(),
+                hideLicenseReport ? null : dataReader[Db2CatalogProjectionColumnNames.LicenseReportUrl]?.ToString(),
+                deprecationInfo);
+        }
+
+        public FeedPackageDeprecationInfo ReadDeprecationInfoFromDataRecord(DbDataReader dataReader)
+        {
+            var deprecationReasons = new List<string>();
+            var deprecationStatus = (PackageDeprecationStatus)dataReader.ReadNullableInt32(Db2CatalogProjectionColumnNames.DeprecationStatus);
+
+            if (deprecationStatus.HasFlag(PackageDeprecationStatus.Legacy))
```

Comment:
Could we use a method on `Enum` for this so that the method does need to be updated for every new `PackageDeprecationStatus` flag?

[Based on this thread from StackOverflow:](https://stackoverflow.com/questions/4171140/iterate-over-values-in-flags-enum)

```
foreach (var deprecationStatusFlag in Enum.GetValues(typeof(PackageDeprecationStatus)))
{
    if (deprecationStatusFlag == PackageDeprecationStatus.NotDeprecated)
    {
        continue;
    }

    if (deprecationStatus.HasFlag(deprecationStatusFlag))
    {
        deprecationReasons.Add(deprecationStatusFlag.ToString());
    }
}
```

Reply:
Fixed!